src/megatron/bridge/data/vlm_datasets/vlm_energon_prodvider.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Any, Optional
+
+from torch import int_repr
+
+from megatron.bridge.data.datasets.energon.base_energon_datamodule import EnergonMultiModalDataModule
+from megatron.bridge.data.utils import DatasetBuildContext, DatasetProvider
+
+
+@dataclass(kw_only=True)
+class EnergonVLMConversationProvider(DatasetProvider):
```

Comment:
why have this be the dataset provider interface? why not make `EnergonMultiModalDataModule` implement `DatasetProvider` so users don't need to keep settings in sync across multiple classes, especially if defining their own?